test/e2e/data/kustomize/k8s-upgrade/upgrade-to-template.yaml

```diff
@@ -0,0 +1,43 @@
+# This is just a copy of the normal cluster template OSMTs.
+# It is needed because we need a static name that we can put in the e2e config.
+# The normal templates has the cluster name as part of the OSMT name.
+# See CONTROL_PLANE_MACHINE_TEMPLATE_UPGRADE_TO and WORKERS_MACHINE_TEMPLATE_UPGRADE_TO
+# in the e2e config.
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
+kind: OpenStackMachineTemplate
+metadata:
+  name: upgrade-to-control-plane
+  labels:
+    clusterctl.cluster.x-k8s.io/move: ""
```

Comment:
This label is important to get the (not yet in use) template moved to the self hosted cluster. Otherwise it would be left behind since it is not part of the owner chain at that point.

Reply:
yes, I think it's something I tried to add into CAPO before, IIRC it's moving stuffs from boostrap to deployed cluster?

Reply:
Exactly. To be clear, it is only needed on this template because it is not part of the cluster owner chain. All resources that are part of the cluster are moved automatically. For example, the OpenStackMachineTemplates that are used when initializing the workload cluster are automatically moved.